data_structures/stacks/balanced_parentheses.py

```diff
@@ -1,23 +1,53 @@
 from .stack import Stack
 
-__author__ = "Omkar Pathak"
 
-
-def balanced_parentheses(parentheses):
-    """ Use a stack to check if a string of parentheses is balanced."""
-    stack = Stack(len(parentheses))
-    for parenthesis in parentheses:
-        if parenthesis == "(":
-            stack.push(parenthesis)
-        elif parenthesis == ")":
-            if stack.is_empty():
+def balanced_parentheses(parentheses: str) -> bool:
+    """Use a stack to check if a string of parentheses is balanced.
+    >>> balanced_parentheses("([]{})")
+    True
+    >>> balanced_parentheses("[()]{}{[()()]()}")
+    True
+    >>> balanced_parentheses("[(])")
+    False
+    >>>
+    """
+    stack = Stack()
+    for bracket in parentheses:
+        if bracket in ("(", "[", "{"):
+            stack.push(bracket)
+        elif bracket in (")", "]", "}"):
+            if stack.is_empty() or not is_paired(stack.pop(), bracket):
                 return False
-            stack.pop()
     return stack.is_empty()
 
 
+def is_paired(left_bracket: str, right_bracket: str) -> bool:
+    """
+    >>> brackets = {"(" : ")", "[" : "]", "{" : "}"}
```

Comment:
Let’s make this dict a global named `bracket_pairs` and then we can eliminate `is_paired()` because it is the same as `bracket_pairs[left_bracket] == right_bracket`